tutorials/bigquery-ibis/ibis_bigquery.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+YOUR_PROJECT_ID = os.environ['GOOGLE_CLOUD_PROJECT']
+
+# [START bigquery_ibis_connect]
+import ibis
+
+conn = ibis.bigquery.connect(
+    project_id=YOUR_PROJECT_ID,
+    dataset_id='bigquery-public-data.stackoverflow')
+# [END bigquery_ibis_connect]
+
+# [START bigquery_ibis_table]
+table = conn.table('posts_questions')
+print(table)
+# BigQueryTable[table]
+# name: bigquery-public-data.stackoverflow.posts_questions
+# schema:
+#   id : int64
+#   title : string
+#   body : string
+#   accepted_answer_id : int64
+#   answer_count : int64
+#   comment_count : int64
+#   community_owned_date : timestamp
+#   creation_date : timestamp
+#   favorite_count : int64
+#   last_activity_date : timestamp
+#   last_edit_date : timestamp
+#   last_editor_display_name : string
+#   last_editor_user_id : int64
+#   owner_display_name : string
+#   owner_user_id : int64
+#   post_type_id : int64
+#   score : int64
+#   tags : string
+#   view_count : int64
+# [END bigquery_ibis_table]
+
+# [START bigquery_ibis_table_not_exist]
+try:
+    doesnt_exist = conn.table('doesnt_exist')
+except Exception as exp:
+    print(str(exp))
+    # Not found: Table bigquery-public-data:stackoverflow.doesnt_exist
+# [END bigquery_ibis_table_not_exist]
+
+# [START bigquery_ibis_table_cross_project]
+reddit_posts_table = conn.table('2018_05', database='fh-bigquery.reddit_posts')
+# [END bigquery_ibis_table_cross_project]
+
+# [START bigquery_ibis_type_error]
+try:
+    table.answer_count.upper()
+except Exception as exp:
+    print(str(exp))
+    # 'IntegerColumn' object has no attribute 'upper'
+# [END bigquery_ibis_type_error]
+
+# [START bigquery_ibis_projection]
+projection = table['creation_date', 'answer_count']
+# [END bigquery_ibis_projection]
+
+# [START bigquery_ibis_transform_timestamp]
+projection = projection.mutate(year=projection.creation_date.year())
+# [END bigquery_ibis_transform_timestamp]
+
+# [START bigquery_ibis_transform_integer]
+has_answer_boolean = projection.answer_count > 0
+# [END bigquery_ibis_transform_integer]
+# [START bigquery_ibis_transform_boolean]
+has_answer_int = has_answer_boolean.ifelse(1, 0)
```

Comment:
Converting a boolean column to an integer column is usually spelled as `has_answer_boolean.cast('int64')`. Is there another reason to prefer this way?

This also has slightly different semantics for `NULL` values [per the docs](https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#conditional-expressions): `cast` preserves `NULL`s, `ifelse` turns thems into `0`.

Reply:
I did an `ifelse()` because it matched the sample query at https://cloud.google.com/bigquery/public-data/stackoverflow#what_is_the_percentage_of_questions_that_have_been_answered_over_the_years

I could change to a cast. A cast probably clearer for this tutorial.